launcher/firstLaunch/firstlaunch_moc.cpp

```diff
@@ -296,12 +296,18 @@ QString FirstLaunchView::getHeroesInstallDir()
 {
 #ifdef VCMI_WINDOWS
 	QString gogPath = QSettings("HKEY_LOCAL_MACHINE\\SOFTWARE\\GOG.com\\Games\\1207658787", QSettings::NativeFormat).value("path").toString();
+	if(!gogPath.isEmpty())
+		return gogPath;
+	gogPath = QSettings("HKEY_LOCAL_MACHINE\\SOFTWARE\\WOW6432Node\\GOG.com\\Games\\1207658787", QSettings::NativeFormat).value("path").toString();
```

Comment:
please don't copy-paste identical suffix

Reply:
@Laserlicht this is not done yet

Reply:
Yes. I've used another approach.

I prefer a data table of exact registry keys over string manipulation / prefix here. IMO it's better, because its the full key you can read out from registry:
![grafik](https://github.com/user-attachments/assets/4d2b1a85-8931-4cbf-b693-66b653ce7a83)

Reply:
With the current approach you still need to replace double backslashes with a single one to put it in regedit, I don't see any advantage.

The proper way here is to use Cartesian product of arrays: you have some prefixes and some suffixes and you want to get all possible combinations. The intent will be clear to anyone reading the code.

We're writing production code here, please avoid any form of copy-paste.

Reply:
> With the current approach you still need to replace double backslashes with a single one to put it in regedit, I don't see any advantage.

But even that's much faster and error resistant than clicking trough the tree. Clicking trough the registry is sometimes painful. :D

I've still another opinion in this specific case (not in general) - I do not want to invest time to implement suggested solution.

Reply:
While kambala is not wrong, such approach would probably result in more complex and less readable code in this particular case.

Unless number of potential locations growth, current approach is fine IMO

Reply:
The suggested solution is indeed complex 😄 

for p in prefixes
  for s in suffixes
    keys += p+s

Reply:
and here is the rest of the owl:
```cpp
QStringList prefixes = {
 "HKEY_LOCAL_MACHINE\\SOFTWARE",
 "HKEY_LOCAL_MACHINE\\SOFTWARE\\WOW6432Node"
};

QVector<QPair<QString, QString>> suffixes = {
	{ "\\GOG.com\\Games\\1207658787",                               "path"    }, // Gog on x86 system
	{ "\\New World Computing\\Heroes of Might and Magic® III\\1.0", "AppPath" }, // H3 Complete on x86 system
	{ "\\New World Computing\\Heroes of Might and Magic III\\1.0",  "AppPath" }, // some localized H3 on x86 system
}

for (const auto & p : prefixes)
{
	for (const auto & s : suffixes)
	{
		QString actualKey = p + s;
		QString path = QSettings(actualKey, QSettings::NativeFormat).value(s.value()).toString();
		if(!path.isEmpty())
			return path;
	}
}
```

Does not looks to be more readable than current version